Chart rendering (#9)
skoudmar/Ros2TraceAnalyzer: README.md

Reviewer: Document how to use the `property` subcommand in README

Reviewer: Update this

```

## Analyze

> This command analyzes traces and saves relevant information for later use into JSON, TXT and DOT files. 

Reviewer 2: ```suggestion
This command analyzes traces and saves relevant information for later use into files. 
```

> The utilization analysis is based solely on timestamps from ROS
> callbacks. It ignores kernel scheduling events such as context
> switches and other activities executed by the application outside of
> callbacks. Therefore, the results are not guaranteed to be always
> correct. However, they are already useful indication for when
> something goes wrong in your application.

## Chart

> Generates a chart of an analysed property for a node. This command can be

Reviewer 2: ```suggestion
Generates a chart of an analysed property for the given entity. This command can be
```

          Increase logging verbosity

  -i, --input <FILENAME>
          Binary bundle file name or a directory containing r2ta_results.sqlite file
          
          [default: r2ta_results.sqlite]

  -q, --quiet...
          Decrease logging verbosity

  -o, --output <FILENAME>
          Directory or filename where to store the chart
          

>           For directories, the chart file name will be <ID>_<QUANTITY>.svg.

Reviewer 2: This seems not to be true. The file name in the example below was: `15_execution_timing_histogram_0_800x600.svg`.

The file name is also printed to stdout. If this is intentional, it should be documented in the description of the command above.

BTW I still think the logic here is unnecessarily difficult. If the parameter to `-o` is just the file name and if `-o` is not given, the image will be sent to stdout, it will be not only easier to implement, but also to document.

          
          For filenames, the output type is determined by its extension. Supported extensions are: SVG [default] and PNG.
          
          If not given, the current directory is used.

      --overwrite
          Overwrite the output file if present

      --quantity <QUANTITY>

>           The value to plot into the chart

Reviewer 2: ```suggestion
          The quantity to plot into the chart
```

          Possible values:
          - callback-duration: Callback execution durations
          - activation-delay:  Delays between callback or timer activations
          - publication-delay: Delays between publisher publications
          - message-delay:     Delays between subscriber messages
          - message-latency:   Latency of a communication channel

      --size <WIDTHxHEIGHT>
          The size of the image in pixels
          
          - For PNG this directly translates to pixels
          
          - For SVG this is the size in pixels with scale 1.0
          
          [default: 800x600]

  -h, --help
          Print help (see a summary with '-h')
```

### Examples

> - Generate a histogram chart of callback durations with default bin count for node ID 15 from and to the current directory with default (600x400) size

Reviewer 2: ```suggestion
- Generate a histogram chart of callback durations for node ID 15 from `r2ta_results.sqlite` in the current directory. The result will appear in the current directory as `15_execution_timing_histogram_0_800x600.svg`
```
The above is the current behavior, which is not the same as documented (see also above)! The file name is `15_execution_timing_histogram_0_800x600.svg`. Why the file name contains `execution_timing` when the command requested `callback-duration`?

```sh
Ros2TraceAnalyzer chart --element-id 15 --quantity callback-duration histogram

> ```

Reviewer 2: Indent the commands by two spaces to be a part of the list item.

- Generate a scatter chart of publication delays for node ID 63 with size 450 by 700 px into file chart.png
```sh
Ros2TraceAnalyzer chart --element-id 63 --quantity publication-delay --size 450x700 -o chart.png scatter
```

- Generate a histogram chart with 52 bins of message latencies from results file located elsewhere into temporary directory with default name and custom size for node ID 124
```sh
Ros2TraceAnalyzer chart --element-id 124 --quantity message-latency --size 641x531 -i /some/file.sqlite -o /tmp histogram --bins 52
```

> 

Reviewer 2: Two examples are probably sufficient. I'd remove this one.

  -h, --help               Print help
```

> The `extract graph` command always extracts the `dependency-graph` stored in the results file.

Reviewer 2: ```suggestion
The `extract graph` command extracts the `dependency-graph` from the results file.
```

> The `extract property` command extracts selected property (analysed value of an element) for the given element id. The possible properties are the same as analysed by the `dependency-graph` analysis. IDs are assigned serially but in no deterministic order (see [ID assignment](#id-assignment)) and are in no way connected to the ROS element. 

Reviewer 2: ```suggestion
The `extract property` command extracts selected property, e.g. message latency, for the given element. The possible properties are the same as analysed by the `dependency-graph` analysis. IDs are assigned consecutively starting from 1, but in no deterministic order (see [ID assignment](#id-assignment)) and are in no way connected to ROS objects.
```

I don't understand "the same as analysed by the `dependency-graph` analysis" - let's talk about it tomorrow.

Reviewer 2: IDs are assigned consecutively starting from 1, in order depending on the order of events in the trace and are in no way connected to ROS objects.
